### ISSUE TITLE
holdout got broken

### ISSUE THREAD

**Issue by petricek** [VowpalWabbit/vowpal_wabbit#260]
In the example below the constant weight gets learned correctly but holdout for some reason reports 0 loss.

To replicate:

```
echo "\
1 |
-1 |
1 |
1 |
-1 |
1 |
1 |
-1 |
1 |
1 |
-1 |
1 |
" | vowpalwabbit/vw --passes 100 -k --cache_file bug.cache 
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
decay_learning_rate = 1
creating cache_file = bug.cache
Reading datafile = 
num sources = 1
average    since         example     example  current  current  current
loss       last          counter      weight    label  predict features
1.000000   1.000000            1         1.0   1.0000   0.0000        1
1.470775   1.941550            2         2.0  -1.0000   0.3934        1
1.103478   0.736181            4         4.0   1.0000   0.2565        1
1.202135   1.300793            8         8.0  -1.0000   0.4195        1
0.000000   0.000000           16        16.0   1.0000   0.2857        1 h
0.000000   0.000000           32        32.0  -1.0000   0.3600        1 h

finished run
number of examples per pass = 12
passes used = 4
weighted example sum = 49
weighted label sum = 13
average loss = 0 h
best constant = 0.288889
total feature number = 49
```
